Tests/ApolloTests/WebSocket/WebSocketTransportTests.swift

```diff
@@ -7,6 +7,17 @@ import Starscream
 class WebSocketTransportTests: XCTestCase {
 
   private var webSocketTransport: WebSocketTransport!
+  static var webSocketProvider: ApolloWebSocketClient.Type = MockWebSocket.self
+
+  override class func setUp() {
+    super.setUp()
+    swap(&webSocketProvider, &WebSocketTransport.provider)
+  }
+
+  override class func tearDown() {
+    super.tearDown()
+    swap(&webSocketProvider, &WebSocketTransport.provider)
```

Comment:
I'm not super familiar with `swap` - what's happening here?

Reply:
From what I can understand of it - WebSocketTransport.provider is [defined as `ApolloWebSocket.self`](https://github.com/apollographql/apollo-ios/blob/fix-watchquery-tests/Sources/ApolloWebSocket/WebSocketTransport.swift#L30) but needs to be mocked out for successful control of the `testUpdateConnectingPayload` test. We can't get in to that class though so `swap` is used as a manual memory change to instead force a `MockWebSocket.self` object to be created [here](https://github.com/apollographql/apollo-ios/blob/fix-watchquery-tests/Sources/ApolloWebSocket/WebSocketTransport.swift#L116). Think of it kinda like Obj-C swizzling - in an abstract way.

Is that correct @AnthonyMDev?

Reply:
Swap just swaps the object at pointer A with the object at Pointer B.

In a few of the tests here we were setting the `WebSocketTransport.provider` to a mock. That property is a `static var`, which means that once you set it, you've changed global state. Unit tests should be isolated and any changes to global state need to be reverted after the tests are done. That wasn't happening.

By swapping them, we store whatever was set in the property before this test suite, and the revert it when the `WebSocketTransportTests` are complete.

Reply:
Having to manually swap pointers is a bit of a code smell for me - it points out that we probably haven't architected this well for testing. 

Is there a way to set this up with either a protocol or a subclass that provides the appropriate implementation? 

I'm open to saying we should make a separate issue and punt on this for now, but we definitely should address places where our testability is kind of sucky before we hit 1.0.

Reply:
Yeah, fully agree. I wanna improve on that before 1.0. This was just a bandaid for now.